src/traces/parcoords/attributes.js

```diff
@@ -0,0 +1,266 @@
+/**
+* Copyright 2012-2016, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var colorAttributes = require('../../components/colorscale/color_attributes');
+var colorbarAttrs = require('../../components/colorbar/attributes');
+
+var extendFlat = require('../../lib/extend').extendFlat;
+
+/*
+todo comments (some of them relating to items that should be unexposed anyway):
+- `domain` should be DRYed up across other plots
+- add attribute to `dimensions` for switching dimensions on/off
+- add attribute to `dimensions` for initial filter domain
+- clarify what the actual use of `_isLinkedToArray: 'dimension'` - esp. the value - is below
+- add attribute for color clamping
+- switch to ploty standard color notation rather than RGB tuple
+- switch to 0..1 for opacity rather than 0..255
+- tie pixelratio to window.devicePixelRatio in `defaults.js`
+- consider if we need a `focusopacity` attribute besides focusalphablending; making settings more symmetric between
+      focus and context
+- hardcode verticalpadding
+- this minor but ergonomic `integerpadding` isn't fully working yet - either finish it or remove it
+*/
+
+module.exports = {
+
+    domain: {
+        x: {
+            valType: 'info_array',
+            role: 'info',
+            items: [
+                {valType: 'number', min: 0, max: 1},
+                {valType: 'number', min: 0, max: 1}
+            ],
+            dflt: [0, 1],
+            description: [
+                'Sets the horizontal domain of this `parcoords` trace',
+                '(in plot fraction).'
+            ].join(' ')
+        },
+        y: {
+            valType: 'info_array',
+            role: 'info',
+            items: [
+                {valType: 'number', min: 0, max: 1},
+                {valType: 'number', min: 0, max: 1}
+            ],
+            dflt: [0, 1],
+            description: [
+                'Sets the vertical domain of this `parcoords` trace',
+                '(in plot fraction).'
+            ].join(' ')
+        }
+    },
+
+    dimensions: {
+        _isLinkedToArray: 'dimension',
+        id: {
+            valType: 'string',
+            role: 'info',
+            description: 'Identifier of a dimension. Must be a unique string across all dimensions.'
+        },
+        label: {
+            valType: 'string',
+            role: 'info',
+            description: 'The shown name of the dimension.'
+        },
+        integer: {
```

Comment:
For `integer` variables, there's

1. No fractional ticks (e.g. there's no `1.5` tick between `1` and `2`
2. Brushing snaps to the integer value (see below)

![integer snapping](https://cloud.githubusercontent.com/assets/1548516/21239854/03c0b978-c309-11e6-9838-c0250ec27e2e.gif)

The reason is that, as per client discussion, in the 1st iteration we'll have no support for `ordinal` and `nominal` dimensions, i.e. such user dimensions will map to integers. But going forward, we should do it with some unified (full-featured Plotly) `axis` approach, with proper `ordinal`, `nominal`, `date` etc. support.